tenderly-suite/role-update-action/role-update/roleUpdate.js

```diff
@@ -0,0 +1,146 @@
+            }
+
+            if (log.topics.length === 0) {
+                throw new Error('INVALID_LOG_TOPICS_LENGTH');
+            }
+
+            if (log.topics[1].length < 26 + 40) {
+                throw new Error('INVALID_LOG_TOPIC_LENGTH');
+            }
+
+            //  account is present in log topic as 32 bytes hex string, with prefixed 0s
+            const account = `0x${log.topics[1].substring(26, 26 + 40)}`;
+
+            let tempSeverity = 1;
+
+            const isTrustedAddress = trustedAddresses.includes(account.toLowerCase());
+
+            if (isTrustedAddress && !isRoleAdded) {
+                tempSeverity = 3;
+            } else if (!isTrustedAddress && isRoleAdded) {
+                tempSeverity = 2;
+            }
+
+            if (isRoleAdded) {
+                roleAddedAccounts.push(account);
+                addedRoles.push(roles);
+                console.log(`Event ${toUtf8Bytes(rolesAdded)} with roles array ${roles} emitted for account ${account}`);
+            } else {
+                roleRemovedAccounts.push(account);
+                removedRoles.push(roles);
+                console.log(`Event ${toUtf8Bytes(rolesRemoved)} with roles array ${roles} emitted for account ${account}`);
+            }
+
+            if (tempSeverity > severity) {
+                severity = tempSeverity;
+            }
+        }
+    }
+
+    if (severity) {
+        try {
+            await axios.post(
+                PAGER_DUTY_ALERT_URL,
+                {
+                    routing_key: await context.secrets.get('PD_ROUTING_KEY'),
+                    event_action: 'trigger',
+                    payload: {
+                        summary,
+                        source: `${chainName}-${event.hash}`,
+                        severity: Severity[severity],
+                        custom_details: {
+                            timestamp: Date.now(),
+                            chain_name: chainName,
+                            trigger_event: event,
+                            rolesUpdated: {
+                                roleAddedAccounts,
+                                addedRoles,
+                                roleRemovedAccounts,
+                                removedRoles,
+                            },
+                        },
+                    },
+                },
+                {
+                    'Content-Type': 'application/json',
+                },
+            );
+        } catch (error) {
+            console.log('PD error status: ', error.response.status);
+            console.log('PD error data: ', error.response.data);
+            throw Error('SENDING_ALERTS_FAILED');
+        }
+    } else {
+        throw new Error('NO_ROLE_UPDATES_DETECTED');
+    }
+};
+
+const Role = {
+    0: 'Distributor',
+    1: 'Operator',
+    2: 'FlowLimiter',
+};
+
+const Severity = {
+    1: 'info',
+    2: 'warning',
+    3: 'critical',
+};
```

Comment:
Consider making an enum file down the line and exporting Role and Severity to be re-used across different monitoring scripts